packages/nimble-blazor/package.json

```diff
@@ -24,6 +22,39 @@
     "invoke-publish:npm": "echo \"noop command to swallow npm args\"",
     "copy-resources": "node build/copyNimbleResources.js"
   },
+  "wireit": {
```

Comment:
I didn't take a close look at the dependency tree and files/outputs for the Blazor build. @atmgrifter00 or @msmithNI could you make sure it looks good?

Reply:
I saw a weird behavior in the Blazor build when running the root `npm run build` after `git clean -fdx` and `npm i`.

1. First build rebuilt everything (makes sense)
2. Second build listed every step as "already fresh" (makes sense) except for nimble-blazor `build:release` and `build:client`, which ran their normal `dotnet` commands again (surprising).
3. Third build showed this status for `build:release` (surprising) but didn't trigger a rebuild: `ℹ️ [build:release] Output files were modified since the previous run.`
4. Subsequent builds showed every step as "already fresh".

This is reproducible on my mac if I clean and install again.

Possibly not a showstopper since it builds correctly and caches eventually, but it would be good to have a theory explaining why.

Reply:
I couldn't reproduce your step 2. For me it went 1, 3, 4. I was able to avoid step 3 by excluding a couple files that were being generated/copied by the build:client command. Can you see if you can still reproduce step 2?

Reply:
After pulling your latest changes, I still see step 2 but no longer see step 3. I wouldn't be surprised if this is a mac thing because of differences in .NET behavior or something. Happy to debug more if you have ideas for me to try. Here's the log in case it's helpful.

```
> @ni/nimble-blazor@11.9.2 build
> wireit

✅ [generate-icons:bundle] Already fresh
✅ [generate-hybrid] Already fresh
✅ [generate-icons] Already fresh
🏃 [build:release] Running command "dotnet build -c Release /p:TreatWarningsAsErrors=true /warnaserror"
MSBuild version 17.6.1+8ffc3fe3d for .NET
  Determining projects to restore...
  Restored /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/NimbleBlazor/NimbleBlazor.csproj (in 83 ms).
  Restored /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Shared/Demo.Shared.csproj (in 123 ms).
  Restored /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Client/Demo.Client.csproj (in 147 ms).
  4 of 7 projects are up-to-date for restore.
  
  > @ni/nimble-blazor@11.9.2 copy-resources
  > node build/copyNimbleResources.js
  
  Destination directory for blazor assets: "/Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/NimbleBlazor/wwwroot"
  NimbleBlazor -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/NimbleBlazor/bin/Release/net6.0/NimbleBlazor.dll
  Demo.Shared -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Shared/bin/Release/net6.0/Demo.Shared.dll
  NimbleBlazor.Tests -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Tests/NimbleBlazor.Tests/bin/Release/net6.0/NimbleBlazor.Tests.dll
  NimbleBlazor.Tests.Acceptance -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Tests/NimbleBlazor.Tests.Acceptance/bin/Release/net6.0/NimbleBlazor.Tests.Acceptance.dll
  Demo.Hybrid -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Hybrid/bin/Release/net6.0-windows/Demo.Hybrid.dll
  Demo.Server -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Server/bin/Release/net6.0/Demo.Server.dll
  Demo.Client -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Client/bin/Release/net6.0/Demo.Client.dll
  Demo.Client (Blazor output) -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Client/bin/Release/net6.0/wwwroot

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.09
✅ [build:release] Executed successfully
🏃 [build:client] Running command "dotnet publish -p:BlazorEnableCompression=false -c Release Examples/Demo.Client --output dist/blazor-client-app"
MSBuild version 17.6.1+8ffc3fe3d for .NET
  Determining projects to restore...
  Restored /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/NimbleBlazor/NimbleBlazor.csproj (in 114 ms).
  Restored /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Shared/Demo.Shared.csproj (in 134 ms).
  Restored /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Client/Demo.Client.csproj (in 171 ms).
  
  > @ni/nimble-blazor@11.9.2 copy-resources
  > node build/copyNimbleResources.js
  
  Destination directory for blazor assets: "/Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/NimbleBlazor/wwwroot"
  NimbleBlazor -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/NimbleBlazor/bin/Release/net6.0/NimbleBlazor.dll
  Demo.Shared -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Shared/bin/Release/net6.0/Demo.Shared.dll
  Demo.Client -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Client/bin/Release/net6.0/Demo.Client.dll
  Demo.Client (Blazor output) -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/Examples/Demo.Client/bin/Release/net6.0/wwwroot
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Demo.Client -> /Users/jesse/Documents/GitHub/nimble2/packages/nimble-blazor/dist/blazor-client-app/
✅ [build:client] Executed successfully
✅ [build] No command to execute
```

Reply:
I don't think the actual build:release and build:client commands should go through wireit caching (should not specify output files).
My understanding is MSBuild has it's own caching / incremental build mechanisms based on timestamps instead of hashes like wireit. I'm not sure how wireit treats timestamps and (mostly guessing) that the two systems won't play well together.